src/os/bluestore/BlueStore.cc

```diff
@@ -1280,9 +1280,8 @@ ostream& operator<<(ostream& out, const BlueStore::SharedBlob& sb)
   return out << ")";
 }
 
-BlueStore::SharedBlob::SharedBlob(uint64_t i, const string& k, Cache *c)
```

Comment:
That would be a bit nicer, yeah.. It's just the open_shared_blob() caller that should use a new ctor, and the new ctor should do the get_shared_blob_key() call